APISamples/typescript/src/home.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { api, CatalogItem, CatalogItemType, User } from "./api";
+import { Link } from "react-router-dom";
+import * as moment from 'moment';
+import { Navbar } from "./navbar";
+
+interface IHomeState {
+    items?: CatalogItem[],
+    user?: User,
+    path: string
+}
+
+export class Home extends React.Component<any,IHomeState> {
+
+    constructor(){
+        super();
+        this.state = { path:'/' };
+    }
+
+    componentDidMount(){
+        api.meAsync().then((user:User)=>this.setState({user}));
+        api.getFolderItemsAsync().then((items:CatalogItem[]) => this.setState({items:items.sort((a,b)=>a.Type.localeCompare(b.Type))}));
+    }
+
+    componentWillReceiveProps(props:any){
+        this.setState({path:props.path.pathname, items:undefined});
+        api.getFolderItemsAsync(props.path.pathname)
+            .then((items:CatalogItem[]) => this.setState({items:items.sort((a,b)=>a.Type.localeCompare(b.Type))}));
+    }
+
+    render() {
+        if(!this.state.items)
+            return (
+                <section className="hero is-warning is-fullheight">
+                    <div className="hero-head">
+                        <Navbar user={this.state.user}/>
+                    </div>
+                    <div className="hero-body">
+                        <div className="container has-text-centered"><h1 className="title">Loading Items...</h1></div>
+                    </div>
+                </section>
+            );
+        
+        return (
+            <div>
+                <section className="hero is-warning is-small">
+                    <div className="hero-head">
+                        <Navbar user={this.state.user}/>
+                    </div>
+                    <div className="hero-body">
+                        <div className="container">
+                            <h1 className="title">
+                                Reports
+                            </h1>
+                        </div>
+                    </div>
+                </section>
+                <section className="section">
+                    <div className="container">
+                        <BreadCrumb path={this.state.path} />
+                        <div className="columns is-multiline">
+                            {this.state.items.map( (item:CatalogItem) => <CatalogItemTile item={item}/>)}
+                        </div>
+                    </div>
+                </section>
+            </div>
+        )
+    }
+}
+
+let CatalogItemTile:React.SFC<{item:CatalogItem}> = ({item}) => {
+    let tagColor;
+
+    switch(item.Type){
+        case CatalogItemType.Report:
```

Comment:
Should we add excel to this? Also what exactly is it doing, looks like icon support but seems odd to use info, warning and error as icons.

Reply:
It just for visual guidance, the 'info','warning' and 'error' tags are just a way of naming the usual 'blue', 'yellow, and 'red' colors